src/Orleans.Streaming/Core/IAsyncObserver.cs

```diff
@@ -56,6 +56,6 @@ public interface IAsyncObserver<in T>
         /// </summary>
         /// <param name="ex">An Exception that describes the error that occurred on the stream.</param>
         /// <returns>A Task that is completed when the close has been accepted.</returns>
-        Task OnErrorAsync(Exception ex);
+        Task OnErrorAsync(Exception ex) => throw ex;
```

Comment:
I'd like to add something to this.

I have seen that the Orleans' audience point-of-view for streams is not very well aligned with how it's actually implemented and/or *indented* to be used. I would argue that *a lot* of people use it more for *decoupling* purposes as opposed to a *throughput pipeline* (which IMO there is nothing wrong with it).

That being said: typically this method will be raised on `QueueCacheMissException` which happens on read from the cache and it doesn't mean anything is wrong other than your cache window being too small, or  processing is taking long(er). I would argue that most people (including me) are *fine* with swallowing those exceptions.

Why I argue against `throw ex`?

Having this default implementation, typically means people will not bother overriding the behavior.
I see 3 types of people/usages for this:

1. Those who don't care about these errors:
`OnErrorAsync(Exception ex) => Task.CompletedTask`

2. Those who care about errors but don't care about cache misses:
```
OnErrorAsync(Exception ex)
{
    if (ex is not QueueCacheMissException) 
    {
         throw ex;
    }
}
```

3.  Those who care about every kind of error:
`OnErrorAsync(Exception ex) => throw ex;`

I have a really hard time beliving that most of the audience belongs to number 3, therefor I'd suggest to have the default implementation `Task.CompletedTask`, and the audiences 2 and 3 can override it accordingly to the needs.

p.s: the same holds for `IAsyncBatchObserver<T>`

Reply:
Thank you for your feedback. Your opinions and analysis make sense. We may need to consider developers who are not particularly familiar with Orleans Streaming. Explicitly handling errors in streaming can avoid misunderstandings, especially for beginners. Therefore, I will adopt your suggestion and remove the default implementation for OnErrorAsync.